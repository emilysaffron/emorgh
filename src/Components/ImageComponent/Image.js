import Grid from "@bbc/psammead-grid";
import Image from "@bbc/psammead-image";
import getGridGroups from "../../Helpers/GetGridGroups";
import styled from "@emotion/styled";

const StyledImage = styled(Image)`
  object-fit: cover;
  width: 62.5%;
  max-height: 62.5%;
  padding-bottom: 1rem;
`;

const Photo = ({ alt, url }) => {
  return (
    <Grid
      item
      startOffset={getGridGroups([2, 2, 2, 2, 2, 2])}
      columns={getGridGroups([8, 8, 8, 8, 8, 8])}
    >
      <StyledImage src={url} alt={alt} />
    </Grid>
  );
};
export default Photo;
