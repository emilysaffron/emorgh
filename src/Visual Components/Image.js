import Grid from "@bbc/psammead-grid";
import Paragraph from "@bbc/psammead-paragraph";
import Image from "@bbc/psammead-image";

import styled from "@emotion/styled";

const StyledImage = styled(Image)`
  object-fit: cover;
  width: 62.5%;
  max-height: 62.5%;
  padding-bottom: 1rem;
`;

const Photo = ({ alt, url }) => {
  return (
    <Grid
      item
      startOffset={{
        group0: 2,
        group1: 2,
        group2: 2,
        group3: 2,
        group4: 2,
        group5: 2,
      }}
      columns={{
        group0: 8,
        group1: 8,
        group2: 8,
        group3: 8,
        group4: 8,
        group5: 8,
      }}
    >
      <StyledImage src={url} alt={alt} />
    </Grid>
  );
};
export default Photo;
