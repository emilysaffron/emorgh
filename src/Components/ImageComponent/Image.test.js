import renderer from "react-test-renderer";
import Image from "./Image";

it("Image Snapshot", () => {
  const tree = renderer.create(<Image alt="image" url="photo" />).toJSON();
  expect(tree).toMatchSnapshot();
});
