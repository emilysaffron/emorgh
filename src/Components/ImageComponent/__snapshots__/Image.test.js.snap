// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Snapshot 1`] = `
<div
  className="css-vxrtof-GridComponent e57qer20"
  dir="ltr"
>
  <img
    alt="image"
    className="css-1rfwwn3-StyledImg e1enwo3v0"
    height={null}
    sizes={null}
    src="photo"
    srcSet={null}
    width={null}
  />
</div>
`;
